### ISSUE TITLE
Add GenBio-PathFM as a supported tile encoder

### ISSUE THREAD

**Issue by clemsgrs** [clemsgrs/slide2vec#211]
## Summary

Add **GenBio-PathFM** ([genbio-ai/genbio-pathfm](https://github.com/genbio-ai/genbio-pathfm), weights [`genbio-ai/genbio-pathfm`](https://huggingface.co/genbio-ai/genbio-pathfm)) as a supported **tile encoder**.

GenBio-PathFM is a large (1.1B-param) histopathology FM reported as the strongest *open-weight* model trained exclusively on public data (JEDI = JEPA+DINO training). Weights are openly downloadable. It loads as a custom HF `AutoModel` (not timm), uses **non-ImageNet** normalization, and needs one output-dimension question resolved before registering — so it is **medium effort**, distinct from the timm-wrapped encoders.

## Model facts (verified against `config.json` + HF card)

- Arch: custom ViT, `model_type="genbio_pathfm"`, **`embed_dim=1536`**, `img_size=224`, `patch_size=16`, `depth=40`, ~1.1B params.
- Load: **`AutoModel.from_pretrained("genbio-ai/genbio-pathfm", trust_remote_code=True)`** (auto_map → `GenBioPathFMModel`). Not a timm model → a custom `TileEncoder` subclass (Titan/PRISM-style), **not** `TimmTileEncoder`.
- Weights: **openly downloadable, not gated** (safetensors).
- Transform: **custom normalization** — `mean=(0.697,0.575,0.728)`, `std=(0.188,0.240,0.187)`, Resize→224, ToTensor. (Not ImageNet — must be set explicitly, like H-optimus does for its custom mean/std.)
- Precision: upstream uses plain PyTorch, no autocast mentioned → `precision="fp32"` (same reasoning as mSTAR, #208). Confirm.
- **Output dim — MUST VERIFY:** `config.json embed_dim` is **1536**, but the HF card advertises a **feature dimension of 4608** (= 3×1536), implying the recommended slide-feature output concatenates representations (e.g. CLS ⊕ mean-patch ⊕ …). The registered `encode_dim` must match what the chosen forward actually returns. **Do not assume — run the model on a dummy tensor and read the shape; pick one canonical output and document it.**

## Implementation

Create `slide2vec/encoders/models/genbio.py` as a custom `TileEncoder`:

```python
"""GenBio-PathFM tile encoder.

GenBio-PathFM (GenBio AI, 2024) is a 1.1B-param ViT histopathology tile encoder
loaded via HF AutoModel (trust_remote_code). Openly downloadable; released under
the GenBio AI Community License.
"""
# - __init__: AutoModel.from_pretrained("genbio-ai/genbio-pathfm", trust_remote_code=True).eval()
# - get_transform(): Resize(224) + Normalize(mean=(0.697,0.575,0.728), std=(0.188,0.240,0.187))
# - encode_tiles(batch): forward -> the chosen canonical embedding (see verification #1)
# - implement encode_dim / device / to() per the SlideEncoder/TileEncoder base contract
#   (mirror a non-timm tile encoder already in the repo, e.g. conch.py / titan.py).

@register_encoder(
    "genbio-pathfm",
    output_variants={"default": {"encode_dim": <VERIFIED_DIM>}},  # 1536 or 4608 — verify
    default_output_variant="default",
    input_size=224,
    supported_spacing_um=0.5,   # VERIFY against paper; default 20x/0.5
    precision="fp32",
    source="genbio-ai/genbio-pathfm",
)
class GenBioPathFM(TileEncoder):
    ...
```

- **Pooled features only, no dense path** in this issue: GenBio is a custom AutoModel, so the `TimmTileEncoder` dense machinery does not apply. Do **not** declare `patch_size` and do **not** add it to the dense coverage lists (dense support is out of scope / a possible follow-up).
- Register `genbio-pathfm` in `slide2vec/encoders/models/__init__.py` (import block + `__all__`, alphabetical).
- Docs: add to `README.md` tile count and the `docs/models.rst` tile table (NOT the dense lists). Note the custom Community License and the non-ImageNet normalization.
- Tests: add `genbio-pathfm` to `tests/test_encoder_registry.py` (`EXPECTED_TILE_ENCODERS`) and `tests/test_regression_core.py` (`list_models("tile")`), plus a focused `test_genbio_pathfm_metadata_contract`. Do **not** add to `test_patch_size_metadata.py` (no dense patch_size). Default suite stays network-free.

## Verification (resolve before finalizing; surface to maintainer if ambiguous)

1. **Output dim & pooling (blocking):** instantiate via `AutoModel(trust_remote_code=True)`, run a `(1,3,224,224)` dummy, and read the output shape. Decide the canonical tile embedding (1536 CLS vs 4608 concat) and set `encode_dim` to match. Document the choice in the docstring. (Heavy/real-weight check → `@pytest.mark.heavy`.)
2. **Spacing/magnification:** the card doesn't state training magnification; default to `0.5` (20×) but confirm against the paper. Flag if it differs.
3. **License:** the **GenBio AI Community License** is custom (not OSI). slide2vec only wraps user-downloaded weights (no redistribution), but read it for acceptable-use/commercial restrictions and note them in the docstring + model docs.
4. **Precision:** confirm upstream inference doesn't use autocast (keep `fp32` if so).

## Acceptance criteria

- [ ] `slide2vec/encoders/models/genbio.py` registers encoder `genbio-pathfm` (level tile, input_size 224, custom normalization, precision fp32, source `genbio-ai/genbio-pathfm`) with `encode_dim` set to the **verified** forward output dim.
- [ ] Loads via HF `AutoModel(trust_remote_code=True)`; `encode_tiles` returns the documented canonical embedding; no dense `patch_size` declared.
- [ ] Metadata-contract tests pass including GenBio-PathFM (`test_encoder_registry.py`, `test_regression_core.py`) with a focused `test_genbio_pathfm_metadata_contract`. Default suite stays network-free.
- [ ] README + `docs/models.rst` list GenBio-PathFM as a tile encoder, noting the custom Community License and non-ImageNet normalization.
